traitlets/config/configurable.py

```diff
@@ -203,8 +210,13 @@ def _config_changed(self, change):
         section_names = self.section_names()
         self._load_config(change.new, traits=traits, section_names=section_names)
 
-    def update_config(self, config):
-        """Update config and load the new values"""
+    def update_config(self, config, skip_env=False):
```

Comment:
When I run:

```python
cfg = Config(key=value)
obj.update_config(cfg)
```

*all* of that configuration should take effect. Not just what hasn't been specified via a particular mechanism. It is not `update_config` in general that should be lower priority than environment variables, but `load_config_files` in particular.

> I always try to have booleans defaulting to False.

I don't think that matters. The name should be the clearest choice and the value should be the logical default behavior. But if you have a clear name where False clearly means "override environment variables", that's okay, too.